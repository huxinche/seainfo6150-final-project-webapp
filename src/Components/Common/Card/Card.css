.card{
    font-family: "TiemposText",Georgia,serif;
    box-shadow: 2px 2px 8px #dedede;
    margin-bottom: 30px;
    /* min-height: 465px; */
    height: auto;
    overflow: auto;
    text-align: center;
}



.card-img{
    position: relative;
    float: left;
    width: 100%;
    overflow: hidden;
}

.card-img img{ 
    height: 465px;
    object-fit: cover;
    width: 100%;
}


.card-content{
    position: relative;
    float: left;
    width: 100%;
    padding: 60px;
    
}
.card-content h3{
    margin-bottom: 30px;
    color: #002760;
    font-weight: 900;
}

.card-content .btn{
    color: white;
}

.card-content p{
   
    font-size: 1.2em;
    line-height: 1.6;
    margin-bottom: 30px;
}

.btn{
    background-color: #cf0000;
    border: 0;
    border-radius: 100px;
    color: #fff;
    display: inline-block;
    font-family: "Circular","Helvetica",sans-serif;
    font-size: inherit;
    line-height: normal;
    margin: 0 15px 15px 0;
    padding: 0.5em 1.3em;
    text-align: center;
    text-decoration: none;
    vertical-align: baseline;
    cursor: pointer;
}

@media only screen and (min-width: 600px) {
    .container{
        padding-right: 50px;
        padding-left: 50px;
    }
    .header{
        height: 100px;
    }
    .logo img{
        width: 100px;
        height: 100px;
    }
 
}

@media only screen and (min-width: 992px) {
    .card{
        text-align: left;
    }
    
    .card-img{
        width: 50%;
    }
    
    .card-content{
        border-top: 10px solid #0078ad;
       width: 50%;
    }

    .vertical .card-img{
        width: 100%;
    }
    .vertical .card-img img{
        max-height: 200px;
    }

    .vertical .card-content{
        border-top: 0px;
        width: 100%;
        text-align: center;
     }


   
}
