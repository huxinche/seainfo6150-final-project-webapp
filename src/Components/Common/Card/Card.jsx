import React from 'react';
import { Link } from 'react-router-dom';
import './Card.css';
import PropTypes from "prop-types";
const Card = (props) => {
    const { image, title, content, link, vertical } = props
    return <div className={+ vertical ? "card vertical" : 'card'}>
        <div className="card-img">
            <img src={image} alt={title} />
        </div>
        <div className="card-content">
            <h3>{title}</h3>
            <p>{content}</p>
            {link && <Link className="btn" to={link}>
                {/* <button  type="button"> */}
                    Learn More
                 {/* </button> */}
            </Link>}
        </div>

    </div>
}

Card.propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    link: PropTypes.string.isRequired,
    divider: PropTypes.bool
}

export default Card;