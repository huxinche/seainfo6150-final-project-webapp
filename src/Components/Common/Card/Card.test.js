import React from "react";
import Card from "./Card";
import image from "../../../images/treefrog.jpg"
it("renders the Banner component correctly", () => {
    const { container } = render(<Card image={image} title={'test'} content={'<p>test content</p>'} link={'/test'}/>);
    expect(container).toMatchSnapshot();
})