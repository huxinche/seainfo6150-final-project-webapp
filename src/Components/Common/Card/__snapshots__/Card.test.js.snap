// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Banner component correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-img"
    >
      <img
        alt="test"
        src="treefrog.jpg"
      />
    </div>
    <div
      class="card-content"
    >
      <h3>
        test
      </h3>
      <p>
        &lt;p&gt;test content&lt;/p&gt;
      </p>
      <a
        class="btn"
        href="/test"
      >
        Learn More
      </a>
    </div>
  </div>
</div>
`;
